pkg/entrypoint/entrypointer.go

```diff
@@ -184,31 +184,36 @@ func (e Entrypointer) Go() error {
 	// strings.Split(..) with an empty string returns an array that contains one element, an empty string.
 	// This creates an error when trying to open the result folder as a file.
 	if len(e.Results) >= 1 && e.Results[0] != "" {
-		if err := e.readResultsFromDisk(); err != nil {
+		if err := e.readResultsFromDisk(pipeline.DefaultResultPath); err != nil {
 			logger.Fatalf("Error while handling results: %s", err)
 		}
 	}
 
 	return err
 }
 
-func (e Entrypointer) readResultsFromDisk() error {
+func (e Entrypointer) readResultsFromDisk(resultDir string) error {
 	output := []v1beta1.PipelineResourceResult{}
 	for _, resultFile := range e.Results {
 		if resultFile == "" {
 			continue
 		}
-		fileContents, err := ioutil.ReadFile(filepath.Join(pipeline.DefaultResultPath, resultFile))
+		fileContents, err := ioutil.ReadFile(filepath.Join(resultDir, resultFile))
 		if os.IsNotExist(err) {
 			continue
 		} else if err != nil {
 			return err
 		}
+		// unmarshal the filesContents to string or array
+		aos := v1beta1.ArrayOrString{}
+		if err := aos.UnmarshalJSON(fileContents); err != nil {
+			return err
+		}
```

Comment:
You can make the new code into a module like `ReadArrayFromFile()`, and just write unit tests on that function

Reply:
Is it used to avoid changing the signature that I added for `readResultsFromDisk`? I think one advantage of that is we can improve the test coverage and I have already added unit tests for that func and cover this code

Reply:
I think here the module is input: file path; output: ArrayOrString, error. This is different from UnmarshalJSON(). I'm also fine with not having making this module and leave these changes to a later PR. This can help you unblock from being influenced from other PRs